taipy/core/taipy.py

```diff
@@ -589,6 +599,9 @@ def get_primary_scenarios(
     """
     scenario_manager = _ScenarioManagerFactory._build_manager()
     scenarios = scenario_manager._get_primary_scenarios()
+
+    if created_start_time or created_end_time:
+        scenarios = scenario_manager._filter_by_creation_time(scenarios, created_start_time, created_end_time)
     if is_sorted:
         scenario_manager._sort_scenarios(scenarios, descending, sort_key)
```

Comment:
Same here. This logic could be delegated to the ScenarioManager.

Reply:
I don't think so. We already have manager methods to handle different options, no complicated logic is handled here